### DISCUSSION TITLE
请问如何插入彩色表格？

### DISCUSSION THREAD

**Discussion by youngfish42** [sjtug/SJTUThesis#1019] [提问] [answered]
**我希望在正文中插入带彩色底纹的表格，但遇到了报错，请问如何处理？**

表格实例如下：（由 [tablesgenerator](https://www.tablesgenerator.com/) 生成）

```
% Please add the following required packages to your document preamble:
% \usepackage{booktabs}
% \usepackage[table,xcdraw]{xcolor}
% Beamer presentation requires \usepackage{colortbl} instead of \usepackage[table,xcdraw]{xcolor}
\begin{table}[]
\centering
\begin{tabular}{@{}lll@{}}
\toprule
C1  & C2                          & C3                        \\ \midrule
0.1 & 2                           & \cellcolor[HTML]{FF8C8C}3 \\
5   & \cellcolor[HTML]{34FF34}0.3 & 4                         \\ \bottomrule
\end{tabular}
\end{table}
```

**我已尝试的解决方案如下：**
在setup文件加载  `\usepackage[table,xcdraw]{xcolor}` ，并在其前面添加 `\PassOptionsToPackage{table,xcdraw}{xcolor}` 修复 xcolor 与 模板不兼容的问题。
![图片](https://github.com/sjtug/SJTUThesis/assets/38998412/041d50d8-db38-453d-87de-d3b50bacca21)

但仍然报错，显示 \cellcolor[HTML] 没有正确闭合。
**报错信息如下图所示：**
![图片](https://github.com/sjtug/SJTUThesis/assets/38998412/87181976-ab44-4861-bab9-1c7620dfcae4) 
![图片](https://github.com/sjtug/SJTUThesis/assets/38998412/57c03d00-4dbd-4ccd-b5be-bc2036b5e3df)

**Comment by LogCreative** [comment, marked as answer]
`sjtuthesis` 文档类会加载 `xcolor` 包，在加载文档类前向宏包传递选项即可：
```latex
\PassOptionsToPackage{table,xcdraw}{xcolor}
\documentclass[type=master]{sjtuthesis}
```

**Reply by youngfish42** [reply]
太感谢了！ 问题已解决~
